lib/APPTurtle/src/Transports/CustomRosTransportTcp.h

```diff
@@ -0,0 +1,153 @@
+/* MIT License
+ *
+ * Copyright (c) 2023 - 2024 Andreas Merkle <web@blue-andi.de>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Custom Micro-ROS transport using TCP over Arduino WifiClient.
+ * @author Norbert Schulz <github@schulznorbert.de>
+ *
+ * @addtogroup Application
+ *
+ * @{
+ */
+#ifndef CUSTOM_ROS_TRANSPORT_TCP_H
+#define CUSTOM_ROS_TRANSPORT_TCP_H
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "CustomRosTransportBase.h"
+
+#include <WiFiClient.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+typedef class CustomRosTransportTcp CustomRosTransport;
+
+/**
+ * Micro-ROS custom transport adaption for TCP.
+ *
+ * TCP protocol uses a record format with a 16 bit length field before the
+ * payload.
+ * 
+ *  +----------------+----------------+-----..........------+
```

Comment:
Table characters are not consistent.

Reply:
It wasn't meant to be a markdown table but a variable sized field. I remove the dots to avoid confusion